[14.0] [FIX] purchase_advance_payment: Wrong residual amt when reconciling credit note

ygol opened the pull request.

ygol commented:
Hi @HviorForgeFlow @MateuGForgeFlow  do you think what I did fix the problem without introducing unwanted behavior?

MiquelRForgeFlow commented:
@ygol some tests are failing

ygol commented:
> @ygol some tests are failing

@MiquelRForgeFlow I've fixed the test but now getting red on coverage and don't know what to do about it. do you think my solution fix the problem? TIA

MiquelRForgeFlow commented:
@ygol The coverage in red means that it needs more tests for the parts of the code which previous tests don't already cover. But the coverage codecov ci is not required, so it is fine as it is now. But you have done several commits. It would be better if you squash them and also put a meaningful message in the commit following the [guidelines](https://www.odoo.com/documentation/16.0/contributing/development/git_guidelines.html#tag-and-module-name) :)

ygol committed:
[FIX] purchase_advance_payment: fix wrong residual amount when reconciling credit note with invoice

When reconciling a bill with a vendor's credit note; the residual amount is erroneous. The vendor credit note amount is subtracted twice.

Fixes #1855

ygol commented:
> @ygol The coverage in red means that it needs more tests for the parts of the code which previous tests don't already cover. But the coverage codecov ci is not required, so it is fine as it is now. But you have done several commits. It would be better if you squash them and also put a meaningful message in the commit following the [guidelines](https://www.odoo.com/documentation/16.0/contributing/development/git_guidelines.html#tag-and-module-name) :)

Thanks for the guidance @MiquelRForgeFlow
all is ok now?

MiquelRForgeFlow reviewed (approved):
LGTM

HviorForgeFlow commented:
ping @rvalyi @clementmbr @PierrickBrun @NathanQj

HviorForgeFlow reviewed (changes requested):
@ygol seems that 87bd4b64934032098ebc3e861a45d5ffc85044dc has more changes that the one specified in the commit regarding FR translation on other modules.

ygol commented:
> @ygol seems that [87bd4b6](https://github.com/OCA/purchase-workflow/commit/87bd4b64934032098ebc3e861a45d5ffc85044dc) has more changes that the one specified in the commit regarding FR translation on other modules.

@HviorForgeFlow I don't understand. what should be done?

HviorForgeFlow commented:
@ygol drop these translation from the commit itself:

![image](https://github.com/OCA/purchase-workflow/assets/32061121/64520e53-5872-43cc-983e-8e4baebede15)

MiquelRForgeFlow commented:
> drop these translation from the commit itself:

Or maybe put them in a different commit.

HviorForgeFlow commented on purchase_advance_payment/__manifest__.py:
```diff
@@ -3,7 +3,7 @@
 {
     "name": "Purchase Advance Payment",
-    "version": "14.0.1.0.1",
+    "version": "14.0.1.1.0",
```
It is not required to update version manually, ocabot will do it.

HviorForgeFlow commented on README.md:
```diff
@@ -34,7 +34,7 @@ addon | version | maintainers | summary
 [purchase_commercial_partner](purchase_commercial_partner/) | 14.0.1.0.0 |  | Add stored related field 'Commercial Supplier' on POs
 [purchase_delivery_split_date](purchase_delivery_split_date/) | 14.0.1.1.1 |  | Allows Purchase Order you confirm to generate one Incoming Shipment for each expected date indicated in the Purchase Order Lines
-[purchase_deposit](purchase_deposit/) | 14.0.1.1.1 |  | Option to create deposit from purchase order
+[purchase_deposit](purchase_deposit/) | 14.0.1.1.2 |  | Option to create deposit from purchase order
```
This change shouldn't be in the commit.

MiquelRForgeFlow replied:
This will be later overwritten by the bot.

ygol replied:
so all is ok for now?

HviorForgeFlow reviewed (changes requested):
Minor comments

HviorForgeFlow commented:
/ocabot merge minor
